backend/canisters/group_index/impl/src/model/public_groups.rs

```diff
@@ -151,13 +149,12 @@ impl PublicGroups {
         self.groups.values()
     }
 
-    pub fn calculate_hot_groups(&self, now: TimestampMillis) -> Vec<ChatId> {
-        let mut rng = StdRng::seed_from_u64(now);
+    pub fn calculate_hot_groups(&self, now: TimestampMillis, random: u32) -> Vec<ChatId> {
         let one_day_ago = now - DAY_IN_MS;
 
         self.iter()
             .filter(|g| !g.is_frozen() && g.has_been_active_since(one_day_ago) && !g.exclude_from_hotlist)
-            .map(|g| (g, rng.next_u32()))
+            .map(|g| (g, random))
```

Comment:
This passes the same value to each group so it isn't really adding any randomness

Reply:
Doh!